File: apps/api/app/services/tickets.py
from datetime import datetime, timezone

from app.schemas.tickets import TicketCreate, TicketResponse


def create_mock_ticket(ticket: TicketCreate) -> TicketResponse:
    return TicketResponse(
        id="ticket_mock_001",
        subject=ticket.subject,
        description=ticket.description,
        priority=ticket.priority,
        status="open",
        created_at=datetime.now(timezone.utc),
    )

File: apps/web/src/lib/tickets.ts
import { type CreateTicketPayload, type Ticket } from "@/types/tickets";

export function formatTicketDate(date: Date) {
  return new Intl.DateTimeFormat("en-US", {
    month: "short",
    day: "numeric",
    year: "numeric",
  }).format(date);
}

export function getNextTicketId(existingTickets: Ticket[]) {
  const ticketNumbers = existingTickets.map((ticket) =>
    Number(ticket.id.replace("TK-", ""))
  );

  return `TK-${Math.max(...ticketNumbers) + 1}`;
}

export function createLocalTicket(
  payload: CreateTicketPayload,
  existingTickets: Ticket[],
  createdAt = new Date()
): Ticket {
  const id = getNextTicketId(existingTickets);

  return {
    id,
    subject: payload.subject,
    customer: payload.customerEmail,
    customerEmail: payload.customerEmail,
    description: payload.description,
    status: "Open",
    priority: payload.priority,
    created: formatTicketDate(createdAt),
  };
}
